daemon/logger/loggerutils/logfile_test.go

```diff
@@ -225,21 +227,15 @@ func TestCheckCapacityAndRotate(t *testing.T) {
 	defer l.Close()
 
 	assert.NilError(t, l.WriteLogEntry(&logger.Message{Line: []byte("hello world!")}))
-
-	dStringer := dirStringer{dir}
-
 	_, err = os.Stat(f.Name() + ".1")
-	assert.Assert(t, os.IsNotExist(err), dStringer)
+	assert.Assert(t, os.IsNotExist(err), dirStringer{dir})
 
 	assert.NilError(t, l.WriteLogEntry(&logger.Message{Line: []byte("hello world!")}))
-	_, err = os.Stat(f.Name() + ".1")
-	assert.NilError(t, err, dStringer)
+	poll.WaitOn(t, checkFileExists(f.Name()+".1.gz"), poll.WithDelay(time.Millisecond), poll.WithTimeout(30*time.Second))
 
 	assert.NilError(t, l.WriteLogEntry(&logger.Message{Line: []byte("hello world!")}))
-	_, err = os.Stat(f.Name() + ".1")
-	assert.NilError(t, err, dStringer)
-	_, err = os.Stat(f.Name() + ".2.gz")
-	assert.NilError(t, err, dStringer)
+	poll.WaitOn(t, checkFileExists(f.Name()+".1.gz"), poll.WithDelay(time.Millisecond), poll.WithTimeout(30*time.Second))
```

Comment:
@thaJeztah you changed `.1` to `.1.gz`

Reply:
Yes this is because compression is async and there is no way for us to know when it will happen.

Reply:
@cpuguy83 I guess I'm not sure I understand what the `.1` was trying to test in the version before this PR.

Reply:
Is it testing the same thing if it's checking for `.1.gz` instead?

Reply:
Just that rotation happens correctly.